packages/neuron-wallet/src/keys/hd.ts

```diff
@@ -0,0 +1,112 @@
+import bip32, { BIP32 } from 'bip32'
+import { Child, KeysData } from './keystore'
+import Address from './address'
+
+enum BIP44Params {
+  Purpose = "44'",
+  // 360 is tentative value
+  CoinTypeTestnet = "360'",
+  Account = "0'",
+}
+
+enum AddressType {
+  Receive = 0,
+  Change = 1,
+}
+
+const MaxAddressNumber = 20
+
+const HD = {
+  generateReceiveAndChangeAddresses: (
+    keysData: KeysData,
+    receiveAddressNumber: number,
+    changeAddressNumber: number,
+  ) => {
+    if (receiveAddressNumber < 1 || changeAddressNumber < 1) {
+      throw new Error('Address number error.')
```

Comment:
Prefer more clear error message.

Reply:
OK

Reply:
You can do this later in another pr.

Reply:
After all reviewing, I will update code one time. 

Maybe it doesn't have more questions.